controllers/cloud.redhat.com/providers/kafka/strimzi.go

```diff
@@ -296,18 +335,88 @@ func (s *strimziProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error
 		}
 	}
 
+	ku := &strimzi.KafkaUser{}
+	nn := types.NamespacedName{
+		Name:      getKafkaUsername(s.Env, app),
+		Namespace: getKafkaNamespace(s.Env),
+	}
+
+	s.Cache.Create(KafkaUser, nn, ku)
+	labeler := utils.GetCustomLabeler(
+		map[string]string{"strimzi.io/cluster": s.Env.Spec.Providers.Kafka.Cluster.Name}, nn, s.Env,
+	)
+
+	labeler(ku)
+
+	ku.Spec = &strimzi.KafkaUserSpec{
+		Authentication: &strimzi.KafkaUserSpecAuthentication{
+			Type: strimzi.KafkaUserSpecAuthenticationTypeScramSha512,
+		},
+		Authorization: &strimzi.KafkaUserSpecAuthorization{
+			Acls: []strimzi.KafkaUserSpecAuthorizationAclsElem{},
+			Type: strimzi.KafkaUserSpecAuthorizationTypeSimple,
+		},
+	}
+
 	// update s.Config.Topics
-	if err := s.processTopics(app); err != nil {
+	if err := s.processTopics(app, ku); err != nil {
 		return err
 	}
 
+	s.Cache.Update(KafkaUser, ku)
+
+	for _, broker := range s.Config.BrokersTLS {
+		if broker.Authtype == nil {
+			continue
+		}
+		if *broker.Authtype == config.BrokerConfigAuthtypeSasl {
+			ku := &strimzi.KafkaUser{}
+			nn := types.NamespacedName{
+				Name:      getKafkaUsername(s.Env, app),
+				Namespace: getKafkaNamespace(s.Env),
+			}
+
+			err := s.Client.Get(s.Ctx, nn, ku)
+			if err != nil {
+				return err
```

Comment:
Do you need to use `errors.Wrap` here to trigger requeue?

Reply:
We shouldn't need to, because any error created by a provider will be cause by the provider and wrapped